monkey/monkey_island/cc/resources/auth/auth.py

```diff
@@ -1,40 +1,67 @@
+import json
+import logging
 from functools import wraps
 
-from flask import abort, current_app
-from flask_jwt import JWT, JWTError, _jwt_required
+import flask_restful
+import flask_jwt_extended
+from flask import make_response, request
+from flask_jwt_extended.exceptions import JWTExtendedException
+from jwt import PyJWTError
 from werkzeug.security import safe_str_cmp
 
 import monkey_island.cc.environment.environment_singleton as env_singleton
 import monkey_island.cc.resources.auth.user_store as user_store
 
-__author__ = 'itay.mizeretz'
+logger = logging.getLogger(__name__)
 
 
 def init_jwt(app):
     user_store.UserStore.set_users(env_singleton.env.get_auth_users())
+    _ = flask_jwt_extended.JWTManager(app)
+    logger.debug("Initialized JWT with secret key that started with " + app.config["JWT_SECRET_KEY"][:4])
 
-    def authenticate(username, secret):
+
+class Authenticate(flask_restful.Resource):
+    """
+    Resource for user authentication. The user provides the username and hashed password and we give them a JWT.
+    See `AuthService.js` file for the frontend counterpart for this code.
+    """
+    @staticmethod
+    def _authenticate(username, secret):
         user = user_store.UserStore.username_table.get(username, None)
         if user and safe_str_cmp(user.secret.encode('utf-8'), secret.encode('utf-8')):
             return user
 
-    def identity(payload):
-        user_id = payload['identity']
-        return user_store.UserStore.user_id_table.get(user_id, None)
-
-    JWT(app, authenticate, identity)
-
+    def post(self):
+        """
+        Example request:
+        {
+            "username": "my_user",
+            "password": "343bb87e553b05430e5c44baf99569d4b66..."
+        }
+        """
+        credentials = json.loads(request.data)
+        # Unpack auth info from request
+        username = credentials["username"]
+        secret = credentials["password"]
+        # If the user and password have been previously registered
+        if self._authenticate(username, secret):
+            access_token = flask_jwt_extended.create_access_token(identity=user_store.UserStore.username_table[username].id)
+            logger.debug(f"Created access token for user {username}: {access_token}")
+            return make_response({"access_token": access_token, "error": ""}, 200)
+        else:
+            return make_response({"error": "Invalid credentials"}, 401)
 
-def jwt_required(realm=None):
-    def wrapper(fn):
-        @wraps(fn)
-        def decorator(*args, **kwargs):
-            try:
-                _jwt_required(realm or current_app.config['JWT_DEFAULT_REALM'])
-                return fn(*args, **kwargs)
-            except JWTError:
-                abort(401)
 
-        return decorator
+# See https://flask-jwt-extended.readthedocs.io/en/stable/custom_decorators/
+def jwt_required(fn):
```

Comment:
This doesn't fit as a replacement? https://flask-jwt-extended.readthedocs.io/en/stable/api/#flask_jwt_extended.jwt_required

Reply:
Raises 500 instead of a more accurate 401 + message.